src/libsync/propagateuploadng.cpp

```diff
@@ -123,21 +250,39 @@ void PropagateUploadFileNG::slotPropfindIterate(const QString &name, const QMap<
     }
 }
 
+void PropagateUploadFileNG::updateRanges(quint64 start, quint64 size)
+{
+    for(auto iter = _serverRanges.begin(); iter != _serverRanges.end(); ++iter)
+    {
+        if (iter->start <= start && iter->size >= size) {
+            iter->start = start + size;
+            iter->size -= size;
+            if (iter->size == 0) {
+                _serverRanges.erase(iter);
+                break;
+            }
+        }
+    }
+}
+
 void PropagateUploadFileNG::slotPropfindFinished()
 {
     auto job = qobject_cast<LsColJob *>(sender());
     slotJobDestroyed(job); // remove it from the _jobs list
     propagator()->_activeJobList.removeOne(this);
 
-    _currentChunk = 0;
```

Comment:
Why `_serverRanges` btw? Isn't this more `_rangesToUpload`?

Reply:
Sure.